### PR TITLE
feat(api): Add save/load mechanism to serialize/deserialize module calibration data.

### PR DESCRIPTION
**Pull request by vegano1**
# Overview

Once the module is calibrated we need to be able to save/load this calibration data so it's re-used on startup and we don't have to calibrate all the time. This pull request adds this mechanism and builds off [RCORE-609](https://opentrons.atlassian.net/browse/RCORE-609)

Closes [RCORE-611](https://opentrons.atlassian.net/browse/RCORE-611)

Module calibration data will be stored in `/data/robot/modules/` with the name `{module_id}.json` where the `module_id` is the serial number of the module, for example

```
root@4e01b6:/data/robot/modules# cat TDV21P20210413B05.json

{"offset": [1.0, 2.0, 3.0], "mount": 1, "slot": 3, "module": "temperatureModuleType", "module_id": "TDV21P20210413B05", "instrument_id": "P50SV3320220729A07", "lastModified": "2023-03-21T14:44:28.062314+00:00", "source": "user", "status": {"markedBad": false, "source": null, "markedAt": null}}
```

# Test Plan

- [x] Make sure we can save module calibration data in `/data/robot/module/{module_id}.json`
- [x] Make sure we can load module calibration data stored in `/data/robot/module/`
- [x] Make sure we can delete module calibration data matching `{module_id}.json` from the disk.
- [x] Make sure we module calibration data is stored when `module_calibration` command is called from OT3repl

# Changelog

- Added loading/saving functions that serialize/deserialize a new ModuleOffsetModel

# Review requests

- Is there anything I'm missing?

[RCORE-609]: https://opentrons.atlassian.net/browse/RCORE-609?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[RCORE-611]: https://opentrons.atlassian.net/browse/RCORE-611?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/src/opentrons/calibration_storage/ot3/models/v1.py [hunk 1]
```diff
@@ -82,3 +84,25 @@ class InstrumentOffsetModel(BaseModel):
     class Config:
         json_encoders = {datetime: lambda obj: obj.isoformat()}
         json_decoders = {datetime: lambda obj: datetime.fromisoformat(obj)}
+
+
+class ModuleOffsetModel(BaseModel):
+    offset: Point = Field(..., description="Module offset found from calibration.")
+    mount: OT3Mount = Field(..., description="The mount used to calibrate this module.")
+    slot: int = Field(..., description="The slot this module was calibrated in.")
```
Comment (Laura-Danielle) [inline api/src/opentrons/calibration_storage/ot3/models/v1.py:92]: Maybe we should also record the instrument id used to calibrate the module as well as the mount?

**Review by Laura-Danielle — APPROVED** [review]
One comment about saving the pipette id used to calibrate the module offsets as well, otherwise it looks good to me.

File: api/src/opentrons/hardware_control/modules/module_calibration.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    # load default if module offset data do not exist
+    module_cal_obj = ModuleCalibrationOffset(
+        slot=slot,
+        offset=Point(*default_module_calibration_offset()),
+        module=module_type,
+        module_id=module_id,
+        source=SourceType.default,
+        status=CalibrationStatus(),
+    )
+    if module_id and ff.enable_ot3_hardware_controller():
```
Comment (sfoster1) [inline api/src/opentrons/hardware_control/modules/module_calibration.py]: Do we really need this feature flag check? On an OT-2, there just won't be module calibration data to load, and that path is already handled by defaulting to 0.

File: api/src/opentrons/hardware_control/modules/module_calibration.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+def save_module_calibration_offset(
+    offset: Point,
+    mount: OT3Mount,
+    slot: int,
+    module: ModuleType,
+    module_id: str,
+    instrument_id: Optional[str] = None,
+) -> None:
+    """Save the calibration offset for a given module."""
+    if module_id and ff.enable_ot3_hardware_controller():
```
Comment (sfoster1) [inline api/src/opentrons/hardware_control/modules/module_calibration.py]: Do we really need this feature flag check? On an OT-2, we just won't be running this flow.

**Review by sfoster1 — APPROVED** [review]
Looks good, but I'd love to remove those feature flag checks - I think there's not actually any bad behavior that they prevent.

**Pull request merged**